x-pack/plugins/infra/server/plugin.ts

```diff
@@ -154,6 +158,19 @@ export class InfraServerPlugin {
     // Telemetry
     UsageCollector.registerUsageCollector(plugins.usageCollection);
 
+    // Custom search strategies
+    plugins.data.search.registerSearchStrategyContext(
+      this.initializerContext.opaqueId,
+      'logSources',
```

Comment:
The compiler rejects this unless the `ISearchContext` has been extended using declaration merging. But then it incorrectly leaks this key to other callers.